specification/nginx/Nginx.Management/ApiKeys.tsp

```diff
@@ -0,0 +1,109 @@
+import "@typespec/rest";
+import "@typespec/http";
+import "@azure-tools/typespec-azure-core";
+import "@azure-tools/typespec-azure-resource-manager";
+
+using Rest;
+using Azure.ResourceManager;
+
+namespace Nginx.NginxPlus;
+
+/**
+ * API key resource returned by the service.
+ */
+@parentResource(NginxDeployment)
+model ApiKey is ProxyResource<ApiKeyResponseProperties> {
+  ...ResourceNameParameter<ApiKey>;
+}
+
+/**
+ * Request payload for creating or updating an API key.
+ */
+@parentResource(NginxDeployment)
+@resource("apiKeys")
+model ApiKeyRequest is ProxyResource<ApiKeyRequestProperties> {
+  ...ResourceNameParameter<ApiKey>;
+}
+
+/**
+ * Properties returned for an API key resource.
+ */
+#suppress "@azure-tools/typespec-azure-resource-manager/arm-resource-provisioning-state" "We intentionally do not include provisioning state for API keys."
+model ApiKeyResponseProperties {
+  /**
+   * Hint for the secret (read-only)
+   */
+  hint: string;
+
+  /**
+   * Expiry of the key
+   */
+  endDateTime: string;
```

Comment:
>string [](http://example.com/codeflow?start=15&length=6)

recommend modeling as utcDateTime

Reply:
This is a resource type from a previous API version. We are just implementing this new API version in Typespec and that's why it's in this diff. 

The only new resource type for this new API version is found in WafPolicies.tsp

Reply:
Makes sense. I just made the documentation clearer. Thanks !

Reply:
And what RFC or whatever are you using for encoding the date time?